eodag/api/constraints.py

```diff
@@ -0,0 +1,165 @@
+    plugin: Union[Search, Api],
+    product_type: str,
+) -> Dict[str, Dict]:
+    """
+    gets the queryables from the constraints using the given parameters
+    For all queryables only values matching the given parameters based on the constraints will be returned
+    :param constraints: list of constraints fetched from the provider
+    :type constraints: list
+    :param params: conditions the constraints should fulfil
+    :type params: dict
+    :param plugin: search or api plugin that is used
+    :type plugin: Union[Search, Api]
+    :param product_type: product type for which the data should be fetched
+    :type product_type: str
+    :returns: dict containing queryable data
+    :rtype: Dict[str, Dict]
+    """
+    constraint_matches = {}
+    params_available = {k: False for k in params.keys()}
+    # check which constraints match the given parameters
+    eodag_provider_key_mapping = {}
+    values_available = {k: set() for k in params.keys()}
+    for i, constraint in enumerate(constraints):
+        params_matched = {k: False for k in params.keys()}
+        for param, value in params.items():
+            provider_key = _get_provider_queryable_key(
+                param, constraint, plugin, product_type
+            )
+            eodag_provider_key_mapping[provider_key] = param
+            if provider_key:
+                params_available[param] = True
+                if value in constraint[provider_key]:
+                    params_matched[param] = True
+                values_available[provider_key].update(constraint[provider_key])
+        constraint_matches[i] = params_matched
+
+    # check if all parameters are available in the constraints
+    for param, available in params_available.items():
+        if not available:
+            raise NotAvailableError(f"parameter {param} is not queryable")
+
+    # add values of constraints matching params
+    queryables = {}
+    for num, matches in constraint_matches.items():
+        if False not in matches.values():
+            for key in constraints[num]:
+                if key in queryables:
+                    queryables[key]["enum"].update(constraints[num][key])
+                elif key not in eodag_provider_key_mapping:
+                    queryables[key] = {}
+                    queryables[key]["enum"] = set(constraints[num][key])
+
+    # check if constraints matching params have been found
+    if len(queryables) == 0:
+        if len(params) > 1:
+            raise NotAvailableError(
+                f"combination of values {str(params)} is not possible"
+            )
+        else:
+            raise NotAvailableError(
+                f"value {list(params.values())[0]} not available for param {list(params.keys())[0]}, "
+                f"possible values: {str(values_available[list(params.keys())[0]])}"
+            )
+
+    return queryables
+
+
+def fetch_constraints(constraints_url: str, plugin: Union[Search, Api]) -> list:
```

Comment:
Typehint incomplete:
- list is an unqualified return type. It should be List[something]
- constraints should not be of type Any